runtime/src/main/java/org/corfudb/runtime/collections/CorfuStreamEntry.java

```diff
@@ -121,4 +121,44 @@ public CorfuStreamEntry(K key, V payload, M metadata, long epoch, long address,
 
         return new CorfuStreamEntry<K, V, M>(key, payload, metadata, epoch, address, operationType);
     }
+
+    /**
+     *  Convert an SMREntry into a CorfuStreamEntry when the SMR arguments have clear types
+     *  (like if called from a commit callback)
+     * @param entry - the entry to convert from.
+     * @param epoch - the epoch this entry was written to/will be written to.
+     */
+    public static
+    CorfuStreamEntry fromSMREntry(@Nonnull SMREntry entry, final long epoch) {
+
+        long address = entry.getGlobalAddress();
+
+        OperationType operationType;
+        switch(entry.getSMRMethod()) {
+            case "put":
+            case "putAll":
+                operationType = OperationType.UPDATE;
+                break;
+            case "clear":
+                operationType = OperationType.CLEAR;
+                break;
+            case "remove":
+                operationType = OperationType.DELETE;
+                break;
+            default:
+                throw new RuntimeException("SMRMethod "+entry.getSMRMethod()
+                        + " cannot be translated to CorfuStreamEntry");
+        }
+        Object[] args = entry.getSMRArguments();
+        if (args.length > 0) {
+            if (args.length > 1) {
+                CorfuRecord record = (CorfuRecord) args[1];
+                return new CorfuStreamEntry((Message)args[0],
+                        record.getPayload(), record.getMetadata(), 0, address, operationType);
+            }
+        }
+
+        return new CorfuStreamEntry((Message)args[0],
```

Comment:
Should this be inside the if (args.length > 0) {} ?

Reply:
yes thanks!